### ISSUE TITLE
The inference result is not good

### ISSUE THREAD

**Issue by yerang823** [e4s2022/e4s#1]
The inference result is not good, is this a model that only works well for trained faces?

Result
source, target, result
<img width="1570" alt="스크린샷 2023-04-21 오전 10 10 10" src="https://user-images.githubusercontent.com/97432613/233517508-08df7000-a5ec-4ec2-909c-7b726ca46598.png">

Furthermore, The results you posted are also different from the result image of the paper.
Result
source, target, my result
<img width="1535" alt="image" src="https://user-images.githubusercontent.com/97432613/233517775-8c9cfb44-3b73-4d04-9437-537ae8775e2e.png">

Here is your result on paper
<img width="432" alt="스크린샷 2023-04-21 오전 1 55 31" src="https://user-images.githubusercontent.com/97432613/233517829-e31b0f76-07fd-464e-8ca7-0c82296d177a.png">

Please provide detailed answers on the above results.

**Comment by e4s2022** [comment]
Hi,  thanks for your posts.

1. The model was trained on aligned faces, did you align you provided example first?

2. The results differ a bit on the bang near the left eyebrow, which is caused by the different operating orders during the mask composition process. You can try to set `hait_first = False` [here](https://github.com/e4s2022/e4s/blob/bc864d3757beb9d1f1027faf26c91d0f5871c2a1/src/utils/swap_face_mask.py#L47).


Hope this can answer your question.

**Comment by yerang823** [comment]
Thank you for your detailed answer.

The problem was that i used not aligned images.
but attribute informations of images are different from target image.
is there any solutions?

![image](https://user-images.githubusercontent.com/97432613/233556120-50a6caf9-71e5-4f51-9eed-cc878e9d2f33.png)

Thank you, in advance.

**Comment by e4s2022** [comment]
Did you refer to the slightly different expression? For this, I recommend trying some recent re-enactment work, e.g., [MegaPortraits: One-shot Megapixel Neural Head Avatars](https://samsunglabs.github.io/MegaPortraits/), 

I will update this repo if I find some better pre-trained models. BTW, any contribution is welcome from the community.

**Comment by yerang823** [comment]
Thank you for your reference link and answer.
And I will wait for your further updates.
Have a nice day !

**Comment by anleleaaa** [comment]
Hello, I want to ask you, how long does it take to change the big concept of the face after loading the initial model, can it be done in real time?

**Comment by BlueWinters** [comment]
> Thank you for your detailed answer.
> 
> The problem was that i used not aligned images. but attribute informations of images are different from target image. is there any solutions?
> 
> ![image](https://user-images.githubusercontent.com/97432613/233556120-50a6caf9-71e5-4f51-9eed-cc878e9d2f33.png)
> 
> Thank you, in advance.

is there any code to show the alignment process?

**Comment by yc-cui** [comment]
> Thank you for your reference link and answer. And I will wait for your further updates. Have a nice day !

Would you be so kind as to update us on any recent progress?
